### DISCUSSION TITLE
Multiple screens

### DISCUSSION THREAD

**Discussion by husnoo** [Bodmer/TFT_eSPI#2508] [Q&A - General]
Hi,

The ESP32 can have two SPI connections, HSPI and VSPI - but it appears the connections have to be hardcoded in the library?

```
The screen controller, interface pins and library configuration settings must be defined inside the library. They can NOT be defined in the Arduino sketch. See the User_Setup_Select.h file for details. This approach has significant advantages, it keeps the examples clean from long configuration options and once the setup is defined any example can be run without modification.
```
Am I reading this correctly? Why is this project the only one to do this, if it's such a significant advantage?

**Comment by Bodmer** [comment]
The library supports 1 screen.

Two screens of the same type can be used on one SPI bus if the chip select is controlled by the sketch.

**Reply by Skywalkerf34** [reply]
hi Bodmer,
thanks for this library
I am facing this situation now need two screen both same chip driver ILI9488 on same SPI , 
an you explain a little (sample code) how to dive the two screen ? 
Rgds

**Reply by danielkboyer** [reply]
https://github.com/kitecraft/Multiple_TFT_Displays

If you're still stuck, check this out.

**Reply by Skywalkerf34** [reply]
Thanks a lot, 
with little code this make it cristal clear !
